### ISSUE TITLE
Error using K.softmax in DeepExplainer

### ISSUE THREAD

**Issue by menajosep** [shap/shap#581]
Hi all,
I'm having an error when using DeepExplainer with my Keras model. The issue comes when I try to use my custom layer:
```
class PredictiveEntropyLayer(Layer):

    def __init__(self, **kwargs):
        self.output_dim = 1
        super(PredictiveEntropyLayer, self).__init__(**kwargs)

    def build(self, input_shape):
        assert isinstance(input_shape, list)
        super(PredictiveEntropyLayer, self).build(input_shape)  # Be sure to call this at the end

    def call(self, x):
        assert isinstance(x, list)
        logits_mu, logits_sigma = x
        noise = K.random_normal((lambda shape: (NUM_PREDICTION_SAMPLES, shape[0], shape[1]))(K.shape(logits_sigma)))
        tiled_logits_mu = K.reshape(K.tile(logits_mu, [NUM_PREDICTION_SAMPLES, 1]), 
                              (lambda shape: (NUM_PREDICTION_SAMPLES, shape[0], shape[1]))(K.shape(logits_mu)))
        tiled_logits_sigma = K.reshape(K.tile(logits_sigma, [NUM_PREDICTION_SAMPLES, 1]), 
                                 (lambda shape: (NUM_PREDICTION_SAMPLES, shape[0], shape[1]))(K.shape(logits_sigma)))
        z = tiled_logits_mu + noise * tiled_logits_sigma
        z_mean = K.mean(z, axis=0)
        print(z_mean)
        sample_probs = K.softmax(z_mean, axis=-1)
        log_probs = K.log(sample_probs+epsilon)
        entropy = K.sum(-sample_probs*log_probs, axis=-1)
        return [tf.reshape(entropy, (lambda shape: (shape[0], self.output_dim))(tf.shape(logits_sigma)))]
      
    def compute_output_shape(self, input_shape):
        assert isinstance(input_shape, list)
        return [input_shape[0]]
```

as the output of the model.
As far as I saw, the error comes from the K.softmax:
```
InvalidArgumentError (see above for traceback): Inputs to operation gradients_25/predictive_entropy_layer_27/Softmax_grad/gradients/gradients_25/predictive_entropy_layer_27/Softmax_grad/truediv_grad/Select_1 of type Select must have the same size and shape.  Input 0: [2000,1] != input 1: [2000,2]
	 [[node gradients_25/predictive_entropy_layer_27/Softmax_grad/gradients/gradients_25/predictive_entropy_layer_27/Softmax_grad/truediv_grad/Select_1 (defined at /usr/local/lib/python3.6/dist-packages/shap/explainers/deep/deep_tf.py:480) ]]
	 [[node gradients_25/embeddings_28/embedding_lookup_grad/Select (defined at /usr/local/lib/python3.6/dist-packages/shap/explainers/deep/deep_tf.py:396) ]]
```
I have even implemented the softmax function by myself, but the error in the true division is the same. Isn't it possible to use the tensorflow softmax with DeepExplainer?

Thanks and best,

Jose Mena.

**Comment by detrin** [comment]
@menajosep Hi, thanks for your issue. To me this seems like an issue on the side of model definition. I remember having similar issue and the solution was to ensure that the dimensionality of Layer has right dimension. In your case it is one, but maybe you could try to reshape it to something like `(shape, 1)`
